infra/Pos/Subscription/Common.hs

```diff
@@ -63,10 +65,14 @@ subscribeTo
     => SendActions m -> NodeId -> m SubscriptionTerminationReason
 subscribeTo sendActions peer = do
     logNotice $ msgSubscribingTo peer
-    outcome <- try $ withConnectionTo sendActions peer $ \_peerData ->
-           pure $ Conversation $ \conv -> do
-               send conv MsgSubscribe
-               recv conv 0 :: m (Maybe Void) -- Other side will never send
+    outcome <- try $ withConnectionTo sendActions peer $ \_peerData -> do
+        pure $ Conversation $ \(conv :: ConversationActions MsgSubscribe Void m) -> do
+            send conv MsgSubscribe
+            forever $ do
+                threadDelay (30 :: Second)
```

Comment:
Ideally we can do better than this. Perhaps it's too complex, but I'd like us to check.

Ideally we delay 'til 30s after the last block was received. Or equivalently, we wait 30s, but reset/postpone our timer each time we receive a block. It doesn't matter if it's easier for us to use receiving a block header as the trigger, or receive anything from the target peer, or applying a block, whatever is easier.